### ISSUE TITLE
App will send SMS over data when recipient no longer has TextSecure installed, no way to revert back to unencrypted messaging

### ISSUE THREAD

**Issue by ghost** [signalapp/Signal-Android#702]
Serious bug that can render conversation with a specific contact almost un-usable. 

Particular case:

My friend had TextSecure, but uninstalled it when he re-flashed his ROM. In the previous version of TextSecure I simply pressed padlock>abort secure session.

However, since the update my phone tries to send him encrypted texts over data. None of these he receives because he does not have the app installed anymore. 

I can long-press send to enable the option to "send unencrypted" however, when sending sms over data is enabled, it will still send encrypted even though I instructed it not to do so. This is another bug. 

The only way to send him texts anymore is to disable sms over data in settings, and then long-press send to send unencrypted, only then will the app actually send an unencrypted text message.  I have to do this every time I want to reply to him. It's greatly annoying. 

So in a nutshell, the three problems are:

1) App does not check that user actually still uses TextSecure, and forces a secure connection even when one cannot be made. 

2) There is no easy way to force the app to send unencrypted texts to a user when data mode is enabled. When data mode is disabled, the method is still unclear and requires "send unencrypted" to be chosen each time you want to send a message. The removal of "abort secure session" has made this problem very bad, I cannot delete old keys from my key inventory. 

3) Long-press the send button when data is enabled to access "send unencrypted" does not work. Will always revert to sending an encrypted message. 

These three problems make the app almost unusable with certain contacts who have uninstalled TextSecure.
